assembly/src/assembler/instruction/procedures.rs

```diff
@@ -91,37 +91,43 @@ impl Assembler {
                     // procedures, such that when we assemble a procedure, all
                     // procedures that it calls will have been assembled, and
                     // hence be present in the `MastForest`.
-                    mast_forest_builder.find_procedure_root(mast_root).unwrap_or_else(|| {
+                    if let Some(root) = mast_forest_builder.find_procedure_root(mast_root) {
+                        root
+                    } else {
                         // If the MAST root called isn't known to us, make it an external
                         // reference.
                         let external_node = MastNode::new_external(mast_root);
-                        mast_forest_builder.ensure_node(external_node)
-                    })
+                        mast_forest_builder.ensure_node(external_node)?
+                    }
```

Comment:
I would probably write this as:
```rust
match mast_forest_builder.find_procedure_root(mast_root) {
    None => {
        // If the MAST root called isn't known to us, make it an external
        // reference.
        let external_node = MastNode::new_external(mast_root);
        mast_forest_builder.ensure_node(external_node)?
    }
    Some(root) => root
}
```
To keep it consistent with how we have things below.